skhep/math/vectors.py

```diff
@@ -545,130 +536,121 @@ def fromiterable(cls, values):
 
     @property
     def x(self):
-        """Return the 3D-vector coordinate x, aka first coordinate at position 0."""
-        return self.__values[0]
+        """Return the Vector3D coordinate x, aka first coordinate at position 0."""
+        return self.__vector3d.x
 
     @x.setter
     def x(self, value):
         """Sets x, aka first coordinate at position 0."""
-        self.__values[0] = value
+        self.__vector3d.x = value
 
     @property
     def y(self):
-        """Return the 3D-vector coordinate x, aka second coordinate at position 1."""
-        return self.__values[1]
+        """Return the Vector3D coordinate x, aka second coordinate at position 1."""
+        return self.__vector3d.y
 
     @y.setter
     def y(self, value):
         """Sets y, aka second coordinate at position 1."""
-        self.__values[1] = value
+        self.__vector3d.y = value
 
     @property
     def z(self):
-        """Return the 3D-vector coordinate z, aka third coordinate at position 2."""
-        return self.__values[2]
+        """Return the Vector3D coordinate z, aka third coordinate at position 2."""
+        return self.__vector3d.z
 
     @z.setter
     def z(self, value):
         """Sets z, aka third coordinate at position 2."""
-        self.__values[2] = value
+        self.__vector3d.z = value
+        
+    @property
+    def vector(self):
+        """Return the Vector3D."""
+        return self.__vector3d.copy()
```

Comment:
Maybe worth making it clear that a copy is returned. Also it's not only that the "Vector3D" is returned but rather the fact that it's the 3D spatial component of the Lorentz vector.

Reply:
Yes agreed!